react/features/recent-list/components/RecentList.native.js

```diff
@@ -0,0 +1,181 @@
+import React from 'react';
+import { ListView, View, TouchableHighlight, Text } from 'react-native';
+import { connect } from 'react-redux';
+import { Icon } from '../../base/font-icons';
+
+import AbstractRecentList from './AbstractRecentList';
+
+import styles, { UNDERLAY_COLOR } from './styles';
+
+/**
+ * The native container rendering the list of the recently joined rooms.
+ *
+ * @extends AbstractRecentList
+ */
+class RecentList extends AbstractRecentList {
+    /**
+     * Initializes a new {@code RecentList} instance.
+     *
+     */
+    constructor() {
+        super();
+
+        this._renderRow = this._renderRow.bind(this);
+        this._renderServerInfo = this._renderServerInfo.bind(this);
+        this._renderConfLength = this._renderConfLength.bind(this);
+        this._onSelect = this._onSelect.bind(this);
+    }
+
+    /**
+     * Implements React's {@link Component#render()}. Renders a list of
+     * recently joined rooms.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+
+        return (
+            <View style = { styles.container }>
+                <ListView
+                    dataSource = { this.state.dataSource }
+                    enableEmptySections = { true }
+                    renderRow = { this._renderRow } />
+            </View>
+        );
+    }
+
+    /**
+    * Renders a single contact list row.
```

Comment:
joined rooms list row?

Reply:
Reworded, I hope that's what you meant :)